gostream/codec/x264/encoder.go

```diff
@@ -22,6 +23,11 @@ type encoder struct {
 // NewEncoder returns an x264 encoder that can encode images of the given width and height. It will
 // also ensure that it produces key frames at the given interval.
 func NewEncoder(width, height, keyFrameInterval int, logger logging.Logger) (ourcodec.VideoEncoder, error) {
+	// Check to make sure dimensions are even.
+	if width%2 != 0 || height%2 != 0 {
+		return nil, errors.New("x264 encoder does not support odd dimensions")
+	}
```

Comment:
If an initial camera resoution is odd - could we change the resolution the same way that we do when we're setting stream options?

Or more explicitly tell the user in this error to return an image that is even dimensions? I think I prefer this option.

Reply:
Because where this is coming from is someone badly encoding a random resolution image.

Reply:
Ok, improved error message here to inform user to provide even dim frames.

> If an initial camera resoution is odd - could we change the resolution the same way that we do when we're setting stream options?

We could do something fancy like this but not sure if it is worth it given that realistically all cameras should be providing even dimension frames. The only time I have seen this is when fake camera allowed odd dimensions.

Reply:
So pete made a camera where he cropped, re-saturated and transformed a bunch of things from an ffmpeg camera and ended up with odd resolutions in the new camera. Remember the return of bytes is really up to the implementer and you can do wild things if you're doing really custom things.

Reply:
Sounsd good. This check should still work in such cases because if there is a size change we re-initialize the encoder and the new resolution (if odd) should hit this.